scripts/partition/bench.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Command-line tool to compare StreamingDataset sample space partitioning performance."""
+
+from argparse import ArgumentParser, Namespace
+from time import time
+
+from streaming.base.partition import get_partitions
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument(
+        '--num_canonical_nodes',
+        type=int,
+        default=8,
+        help='Number of canonical nodes',
+    )
+    args.add_argument(
+        '--num_nodes',
+        type=int,
+        default=8,
+        help='Number of physical nodes',
+    )
+    args.add_argument(
+        '--ranks_per_node',
+        type=int,
+        default=8,
+        help='Ranks per node',
+    )
+    args.add_argument(
+        '--workers_per_rank',
+        type=int,
+        default=8,
+        help='Workers per rank',
+    )
+    args.add_argument(
+        '--batch_size',
+        type=int,
+        default=16,
+        help='Batch size per rank',
+    )
+    args.add_argument(
+        '--sample_in_epoch',
+        type=int,
+        default=0,
+        help='Sample offset in epoch',
+    )
+    args.add_argument(
+        '--min_power',
+        type=int,
+        default=10,
+        help='Minimum dataset size as a power of 2',
+    )
+    args.add_argument(
+        '--max_power',
+        type=int,
+        default=34,
+        help='Maximum dataset size as a power of 2',
+    )
+    args.add_argument(
+        '--power_interval',
+        type=int,
+        default=4,
+        help='Dataset size step as a power of 2',
+    )
+    args.add_argument(
+        '--timeout',
+        type=float,
+        default=120,
+        help='Maximum time to wait for partitioning to complete',
+    )
+    return args.parse_args()
+
+
+def main(args: Namespace) -> None:
```

Comment:
Does this script saves Partition benchmarking results to a file ?

Reply:
Also, we don't currently have another partitioning algo other than `orig` right? just wanted to check

Reply:
Yes, we write the results to /dev/stdout in this same method for maximum compatibility with Unix tooling

Reply:
We (james) have experimented with alternative partitioning algorithms in the past, for which this code was originally written, hence this works across algos